---
title: "StressNet figure supplementary"
output:
  html_document:
    fig_caption: yes
    highlight: zenburn
    self_contained: yes
    theme: cerulean
    toc: yes
    toc_depth: 3
    toc_float: yes
    code_folding: hide
  beamer_presentation:
    colortheme: dolphinF
    fig_caption: no
    fig_height: 5
    fig_width: 5
    fonttheme: structurebold
    highlight: default
    incremental: no
    keep_tex: no
    slide_level: 1
    toc: yes
  slidy_presentation:
    highlight: default
    incremental: no
    smart: no
    slide_level: 1
    self_contained: yes
    keep_md: yes
    smaller: yes
    theme: cerulean
    toc: yes
    widescreen: yes
  ioslides_presentation:
    highlight: zenburn
    incremental: no
  pdf_document:
    fig_caption: yes
    highlight: zenburn
    toc: yes
    toc_depth: 3
  revealjs::revealjs_presentation:
    theme: night
    transition: none
    self_contained: true
    slide_level: 1
    css: ../slides.css
font-import: http://fonts.googleapis.com/css?family=Risque
font-family: Garamond
transition: linear
editor_options: 
  chunk_output_type: console
---



```{r setup}
library(knitr)
knitr::opts_chunk$set(
  fig.width = 15, fig.height = 15, 
  fig.align = "center", 
  size = "tiny", 
  echo = TRUE, eval = TRUE, 
  warning = FALSE, message = FALSE, 
  results = TRUE, comment = "")
options(scipen = 12) ### Max number of digits for non-scientific notation
```

# data import

```{r data import,warning=FALSE,message=FALSE}
library("reshape2")
library("agricolae")
library("ggplot2")
library("ggrepel")
library('gtools')
library('circlize')
library("ComplexHeatmap")

library("tibble")
library("stringr")
library("ggpubr")
library("kableExtra")
library("gridExtra")
library("tidyr")
library("dplyr")
library("patchwork")
library("plyr")
library('cowplot')

```

 
```{r  import}

#phenoscope-data
Geno= c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")
colorCond= c("blue","orange","purple","red")
Cond= c("W+N+","W-N+","W+N-","W-N-")


GrowthParameter <- read.csv('data/Stressnet_Growth_parameter.csv', header = TRUE )
GrowthParameter$Geno <- gsub("\\.0","-0",GrowthParameter$Geno)


PhysiologicalParameter <- read.csv('data/Stressnet_Physiological_parameter.csv', header = TRUE )

PhysiologicalParameter$Geno <- gsub("\\.0","-0",PhysiologicalParameter$Geno)
 
 
```
## Figure_S1
### Figure_S1A : Compactness

```{r Figure_S1A}

subPhenoData <- melt(GrowthParameter[,grepl("Geno|Cond|CompactnessPCD", colnames(GrowthParameter))],id = c("Geno","Cond"))

subPhenoData$Day <- gsub("CompactnessPCD","",subPhenoData$variable)
subPhenoData$Day <- as.numeric(as.character(subPhenoData$Day))
subPhenoData[,"DAS"] <- subPhenoData$Day + 8
 

#### CompactnessPCD normalized by control

µdata <- ddply(subPhenoData,.(Cond,Geno,Day),summarise, CompactnessPCinstant = mean(value,na.rm = T))

# Calculation of the ratio
allµdata <- NULL
µdata[,"Ratio"]<- 1
subsetµdata <- µdata
 for (i in 1:nrow(subsetµdata)) {
 subsetµdata$Ratio[i] =  subsetµdata$CompactnessPCinstant[i]/subsetµdata$CompactnessPCinstant[which(subsetµdata$Day== subsetµdata$Day[i]&subsetµdata$Cond=="W+N+"& subsetµdata$Geno==subsetµdata$Geno[i])]
 }
   allµdata<-rbind(subsetµdata,allµdata)

## Graphe for each genotype (with the ratio above and the significance below) :

Allpval =NULL
for (i in c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) {
dataCol <- subPhenoData[subPhenoData$Geno==i,]
Pval <- data.frame(day=c(1:21))
for (d in 1:21){
		Pval$A[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N+"),]))$"Pr(>F)"[1])
		Pval$B[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W+N-"),]))$"Pr(>F)"[1])
		Pval$C[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N-"),]))$"Pr(>F)"[1])
  }
   Pval[,"Geno"] <- i
  Allpval <- rbind(Pval,Allpval)
}

allµdata[,"DAS"] <- allµdata$Day + 8
Allpval[,"DAS"] <- Allpval$day + 8
allµdata$Geno <- factor(allµdata$Geno,levels= Geno)
Allpval$Geno <- factor(Allpval$Geno,levels= Geno)
#kable(allµdata[allµdata$DAS==31,]) %>% kable_styling() %>% scroll_box(height = "500px")

g1<- ggplot(allµdata,aes(x=DAS,y=Ratio,color=Cond)) + geom_line() + facet_grid(.~Geno,scales = "fixed")  + annotate("segment",x=10,xend=31,y=1,yend=1) +ylab( "Stress/Control Ratio") +  theme(legend.position="top")+ labs(color='Condition')+ scale_colour_manual(values=c("blue","orange","purple","red"),limits=c("W+N+","W-N+","W+N-","W-N-")) + ggtitle("Compactness") +xlab('')

g2 <- ggplot(Allpval,aes(x=DAS))+ geom_line(aes(y=A),colour="orange")+ geom_line(aes(y=B),colour="purple")+ facet_grid(.~Geno,scales = "fixed")+ geom_line(aes(y=C),colour="red")+ ylab("-log(Stress/Control significance)") + xlab("DAS") +  annotate("segment",x=10,xend=31,y=-log(0.05),yend=-log(0.05),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.01),yend=-log(0.01),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.001),yend=-log(0.001),lty=2) 

Figure_S1A <- g1/g2
Figure_S1A

```

### Figure_S1B : Rosette color Hue

```{r Figure_S1B}

subPhenoData <- melt(GrowthParameter[,grepl("Geno|Cond|Hue_newD", colnames(GrowthParameter))],id = c("Geno","Cond"))

subPhenoData$Day <- gsub("Hue_newD","",subPhenoData$variable)
subPhenoData$Day <- as.numeric(as.character(subPhenoData$Day))
subPhenoData[,"DAS"] <- subPhenoData$Day + 8
 

#### Hue_newD normalized by control

µdata <- ddply(subPhenoData,.(Cond,Geno,Day),summarise, RERinstant = mean(value,na.rm = T))

# Calculation of the ratio
allµdata <- NULL
µdata[,"Ratio"]<- 1
subsetµdata <- µdata
 for (i in 1:nrow(subsetµdata)) {
 subsetµdata$Ratio[i] =  subsetµdata$RERinstant[i]/subsetµdata$RERinstant[which(subsetµdata$Day== subsetµdata$Day[i]&subsetµdata$Cond=="W+N+"& subsetµdata$Geno==subsetµdata$Geno[i])]
 }
   allµdata<-rbind(subsetµdata,allµdata)

## Graphe for each genotype (with the ratio above and the significance below) :

Allpval =NULL
for (i in c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) {
dataCol <- subPhenoData[subPhenoData$Geno==i,]
Pval <- data.frame(day=c(1:21))
for (d in 1:21){
		Pval$A[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N+"),]))$"Pr(>F)"[1])
		Pval$B[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W+N-"),]))$"Pr(>F)"[1])
		Pval$C[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N-"),]))$"Pr(>F)"[1])
  }
   Pval[,"Geno"] <- i
  Allpval <- rbind(Pval,Allpval)
}

allµdata[,"DAS"] <- allµdata$Day + 8
Allpval[,"DAS"] <- Allpval$day + 8
allµdata$Geno <- factor(allµdata$Geno,levels= Geno)
Allpval$Geno <- factor(Allpval$Geno,levels= Geno)
#kable(allµdata[allµdata$DAS==31,]) %>% kable_styling() %>% scroll_box(height = "500px")

g1<- ggplot(allµdata,aes(x=DAS,y=Ratio,colour=Cond)) + geom_line() + facet_grid(.~Geno,scales = "fixed")  + annotate("segment",x=10,xend=31,y=1,yend=1) +ylab( "Stress/Control Ratio") +  theme(legend.position="top")+ scale_colour_manual(values=c("blue","orange","purple","red"),limits=c("W+N+","W-N+","W+N-","W-N-")) + ggtitle("Rosette color Hue") +xlab('')+  labs(colour='Condition')

g2 <- ggplot(Allpval,aes(x=DAS))+ geom_line(aes(y=A),colour="orange")+ geom_line(aes(y=B),colour="purple")+ facet_grid(.~Geno,scales = "fixed")+ geom_line(aes(y=C),colour="red")+ ylab("-log(Stress/Control significance)") + xlab("DAS")+  annotate("segment",x=10,xend=31,y=-log(0.05),yend=-log(0.05),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.01),yend=-log(0.01),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.001),yend=-log(0.001),lty=2) 

Figure_S1B <- g1/g2
Figure_S1B

```

### Figure_S1C : Rosette color Saturation

```{r Figure_S1C}

subPhenoData <- melt(GrowthParameter[,grepl("Geno|Cond|Saturation_newD", colnames(GrowthParameter))],id = c("Geno","Cond"))

subPhenoData$Day <- gsub("Saturation_newD","",subPhenoData$variable)
subPhenoData$Day <- as.numeric(as.character(subPhenoData$Day))
subPhenoData[,"DAS"] <- subPhenoData$Day + 8
 

#### Hue_newD normalized by control

µdata <- ddply(subPhenoData,.(Cond,Geno,Day),summarise, RERinstant = mean(value,na.rm = T))

# Calculation of the ratio
allµdata <- NULL
µdata[,"Ratio"]<- 1
subsetµdata <- µdata
 for (i in 1:nrow(subsetµdata)) {
 subsetµdata$Ratio[i] =  subsetµdata$RERinstant[i]/subsetµdata$RERinstant[which(subsetµdata$Day== subsetµdata$Day[i]&subsetµdata$Cond=="W+N+"& subsetµdata$Geno==subsetµdata$Geno[i])]
 }
   allµdata<-rbind(subsetµdata,allµdata)

## Graphe for each genotype (with the ratio above and the significance below) :

Allpval =NULL
for (i in c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) {
dataCol <- subPhenoData[subPhenoData$Geno==i,]
Pval <- data.frame(day=c(1:21))
for (d in 1:21){
		Pval$A[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N+"),]))$"Pr(>F)"[1])
		Pval$B[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W+N-"),]))$"Pr(>F)"[1])
		Pval$C[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N-"),]))$"Pr(>F)"[1])
  }
   Pval[,"Geno"] <- i
  Allpval <- rbind(Pval,Allpval)
}

allµdata[,"DAS"] <- allµdata$Day + 8
Allpval[,"DAS"] <- Allpval$day + 8
allµdata$Geno <- factor(allµdata$Geno,levels= Geno)
Allpval$Geno <- factor(Allpval$Geno,levels= Geno)
#kable(allµdata[allµdata$DAS==31,]) %>% kable_styling() %>% scroll_box(height = "500px")

g1<- ggplot(allµdata,aes(x=DAS,y=Ratio,colour=Cond)) + geom_line() + facet_grid(.~Geno,scales = "fixed")  + annotate("segment",x=10,xend=31,y=1,yend=1) +ylab( "Stress/Control Ratio") +  theme(legend.position="top")+ scale_colour_manual(values=c("blue","orange","purple","red"),limits=c("W+N+","W-N+","W+N-","W-N-")) + ggtitle("Rosette color Saturation") +xlab('')+labs(colour='Condition')


g2 <- ggplot(Allpval,aes(x=DAS))+ geom_line(aes(y=A),colour="orange")+ geom_line(aes(y=B),colour="purple")+ facet_grid(.~Geno,scales = "fixed")+ geom_line(aes(y=C),colour="red")+ ylab("-log(Stress/Control significance)") + xlab("DAS")+  annotate("segment",x=10,xend=31,y=-log(0.05),yend=-log(0.05),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.01),yend=-log(0.01),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.001),yend=-log(0.001),lty=2) 

Figure_S1C <- g1/g2
Figure_S1C


```

### Figure_S1D : Rosette color Value

```{r Figure_S1D}

subPhenoData <- melt(GrowthParameter[,grepl("Geno|Cond|Value_newD", colnames(GrowthParameter))],id = c("Geno","Cond"))

subPhenoData$Day <- gsub("Value_newD","",subPhenoData$variable)
subPhenoData$Day <- as.numeric(as.character(subPhenoData$Day))
subPhenoData[,"DAS"] <- subPhenoData$Day + 8
 

#### Value normalized by control

µdata <- ddply(subPhenoData,.(Cond,Geno,Day),summarise, RERinstant = mean(value,na.rm = T))

# Calculation of the ratio
allµdata <- NULL
µdata[,"Ratio"]<- 1
subsetµdata <- µdata
 for (i in 1:nrow(subsetµdata)) {
 subsetµdata$Ratio[i] =  subsetµdata$RERinstant[i]/subsetµdata$RERinstant[which(subsetµdata$Day== subsetµdata$Day[i]&subsetµdata$Cond=="W+N+"& subsetµdata$Geno==subsetµdata$Geno[i])]
 }
   allµdata<-rbind(subsetµdata,allµdata)

## Graphe for each genotype (with the ratio above and the significance below) :

Allpval =NULL
for (i in c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) {
dataCol <- subPhenoData[subPhenoData$Geno==i,]
Pval <- data.frame(day=c(1:21))
for (d in 1:21){
		Pval$A[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N+"),]))$"Pr(>F)"[1])
		Pval$B[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W+N-"),]))$"Pr(>F)"[1])
		Pval$C[Pval$day==d] <- -log(anova(lm(value~Cond,dataCol[dataCol$Day==d&dataCol$Cond%in%c("W+N+","W-N-"),]))$"Pr(>F)"[1])
  }
   Pval[,"Geno"] <- i
  Allpval <- rbind(Pval,Allpval)
}

allµdata[,"DAS"] <- allµdata$Day + 8
Allpval[,"DAS"] <- Allpval$day + 8
allµdata$Geno <- factor(allµdata$Geno,levels= Geno)
Allpval$Geno <- factor(Allpval$Geno,levels= Geno)
#kable(allµdata[allµdata$DAS==31,]) %>% kable_styling() %>% scroll_box(height = "500px")

g1<- ggplot(allµdata,aes(x=DAS,y=Ratio,colour=Cond)) + geom_line() + facet_grid(.~Geno,scales = "fixed")  + annotate("segment",x=10,xend=31,y=1,yend=1) +ylab( "Stress/Control Ratio") +  theme(legend.position="top")+ scale_colour_manual(values=c("blue","orange","purple","red"),limits=c("W+N+","W-N+","W+N-","W-N-")) + ggtitle("Rosette color Value") +xlab('')+ labs(colour='Condition')


g2 <- ggplot(Allpval,aes(x=DAS))+ geom_line(aes(y=A),colour="orange")+ geom_line(aes(y=B),colour="purple")+ facet_grid(.~Geno,scales = "fixed")+ geom_line(aes(y=C),colour="red")+ ylab("-log(Stress/Control significance)") + xlab("DAS")+  annotate("segment",x=10,xend=31,y=-log(0.05),yend=-log(0.05),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.01),yend=-log(0.01),lty=2) + annotate("segment",x=10,xend=31,y=-log(0.001),yend=-log(0.001),lty=2)
Figure_S1D <- g1/g2
Figure_S1D


```


### Figure_S1E : Water Content


```{r Figure_S1E}

data_all <- PhysiologicalParameter 

Hsdcol <-   HSD.test(aov(WaterContent ~  Cond, data = data_all[ data_all$Geno == "Col-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()  %>% mutate(Geno = "Col-0")
Hsdsvi <-   HSD.test(aov(WaterContent ~  Cond, data = data_all[  data_all$Geno == "Cvi-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Cvi-0")
Hsdsha <-   HSD.test(aov(WaterContent ~  Cond, data = data_all[ data_all$Geno == "Sha",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Sha")
Hsdbur <-   HSD.test(aov(WaterContent ~  Cond, data = data_all[  data_all$Geno == "Bur-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Bur-0")
Hsdtsu <-   HSD.test(aov(WaterContent ~  Cond, data = data_all[  data_all$Geno == "Tsu-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno ="Tsu-0")

Hsd <- rbind(Hsdcol,Hsdsvi,Hsdsha,Hsdbur,Hsdtsu)
Hsd$Geno <- factor(Hsd$Geno,levels= Geno) 
sum <- ddply(data_all,.(Cond,Geno),summarise,n= length(WaterContent[!is.na(WaterContent)]))
sum$Geno <- factor(sum$Geno,levels= Geno) 
data_all$Geno <- factor(data_all$Geno,levels= Geno) 
data_all$Cond <- factor(data_all$Cond,levels= Cond) 

Figure_S1E <- ggboxplot(data_all, x = "Cond", y = "WaterContent", color = "Cond",  palette =c("blue", "orange", "purple","red") ,outlier.shape = "" ,add = "jitter") + ggtitle("Water Content") + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed")  + ylab("W (%)") + theme_bw() +
geom_text(data=Hsd,aes(x=Cond,y=0.98*min(WaterContent),label=groups),vjust=0) + theme(axis.text.x = element_text(angle=90))  + xlab('')+ theme(legend.position="None") + labs(colour='Condition')
Figure_S1E


 
```

### Figure_S1F: Nitrogen content

```{r Figure_S1F}

data_all <-  PhysiologicalParameter

Hsdcol <-   HSD.test(aov(X.N ~  Cond, data = data_all[  data_all$Geno == "Col-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()  %>% mutate(Geno = "Col-0")
Hsdsvi <-   HSD.test(aov(X.N ~  Cond, data = data_all[ data_all$Geno == "Cvi-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Cvi-0")
Hsdsha <-   HSD.test(aov(X.N ~  Cond, data = data_all[  data_all$Geno == "Sha",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Sha")
Hsdbur <-   HSD.test(aov(X.N ~  Cond, data = data_all[  data_all$Geno == "Bur-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Bur-0")
Hsdtsu <-   HSD.test(aov(X.N ~  Cond, data = data_all[ data_all$Geno == "Tsu-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno ="Tsu-0")

Hsd <- rbind(Hsdcol,Hsdsvi,Hsdsha,Hsdbur,Hsdtsu)
Hsd$Geno <- factor(Hsd$Geno,levels= Geno) 

sum <- ddply(data_all,.(Cond,Geno),summarise,n= length(X.N[!is.na(X.N)]))
sum$Geno <- factor(sum$Geno,levels= Geno) 
data_all$Geno <- factor(data_all$Geno,levels= Geno) 
data_all$Cond <- factor(data_all$Cond,levels= Cond) 

Figure_S1F <- ggboxplot(data_all, x = "Cond", y = "X.N", color = "Cond",  palette =c("blue", "orange", "purple","red") ,add = "jitter") +  ggtitle("Nitrogen Content") + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed")  + ylab("N(%)") +  theme_bw() +
geom_text(data=Hsd,aes(x=Cond,y=0.98*min(X.N),label=groups),vjust=0) + theme(axis.text.x = element_text(angle=90))  + xlab('')+ theme(legend.position="None") + labs(colour='Condition') 

Figure_S1F


```

### Figure_S1G: Carbon content

 
```{r Figure_S1G}

data_all <-  PhysiologicalParameter

 
Hsdcol <-   HSD.test(aov(X.C ~  Cond, data = data_all[  data_all$Geno == "Col-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()  %>% mutate(Geno = "Col-0")
Hsdsvi <-   HSD.test(aov(X.C ~  Cond, data = data_all[data_all$Geno == "Cvi-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Cvi-0")
Hsdsha <-   HSD.test(aov(X.C ~  Cond, data = data_all[data_all$Geno == "Sha",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Sha")
Hsdbur <-   HSD.test(aov(X.C ~  Cond, data = data_all[data_all$Geno == "Bur-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Bur-0")
Hsdtsu <-   HSD.test(aov(X.C ~  Cond, data = data_all[ data_all$Geno == "Tsu-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno ="Tsu-0")

Hsd <- rbind(Hsdcol,Hsdsvi,Hsdsha,Hsdbur,Hsdtsu)
Hsd$Geno <- factor(Hsd$Geno,levels= Geno) 

sum <- ddply(data_all,.(Cond,Geno),summarise,n= length(X.C[!is.na(X.C)]))
sum$Geno <- factor(sum$Geno,levels= Geno) 
data_all$Geno <- factor(data_all$Geno,levels= Geno) 
data_all$Cond <- factor(data_all$Cond,levels= Cond) 


Figure_S1G  <- ggboxplot(data_all, x = "Cond", y = "X.C", color = "Cond",  palette =c("blue", "orange", "purple","red") ,add = "jitter") +  ylab("C(%)") + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed")   + ggtitle("Carbon Content") + theme_bw()+ geom_text(data=Hsd,aes(x=Cond,y=0.85*min(X.C),label=groups),vjust=0)+ theme(axis.text.x = element_text(angle=90))  +   xlab('')+ theme(legend.position="None")+ labs(colour='Condition')
Figure_S1G


```

### Figure_S1H: Nitrate Content

```{r Figure_S1H}

data_all <-  PhysiologicalParameter

Hsdcol <-   HSD.test(aov(NitrateContent.μmols.gFW. ~  Cond, data = data_all[  data_all$Geno == "Col-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()  %>% mutate(Geno = "Col-0")
Hsdsvi <-   HSD.test(aov(NitrateContent.μmols.gFW. ~  Cond, data = data_all[  data_all$Geno == "Cvi-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Cvi-0")
Hsdsha <-   HSD.test(aov(NitrateContent.μmols.gFW. ~  Cond, data = data_all[ data_all$Geno == "Sha",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Sha")
Hsdbur <-   HSD.test(aov(NitrateContent.μmols.gFW. ~  Cond, data = data_all[  data_all$Geno == "Bur-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Bur-0")
Hsdtsu <-   HSD.test(aov(NitrateContent.μmols.gFW. ~  Cond, data = data_all[  data_all$Geno == "Tsu-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno ="Tsu-0")

Hsd <- rbind(Hsdcol,Hsdsvi,Hsdsha,Hsdbur,Hsdtsu)
Hsd$Geno <- factor(Hsd$Geno,levels= Geno) 
sum <- ddply(data_all,.(Cond,Geno),summarise,n= length(NitrateContent.μmols.gFW.[!is.na(NitrateContent.μmols.gFW.)]))
sum$Geno <- factor(sum$Geno,levels= Geno) 
data_all$Geno <- factor(data_all$Geno,levels= Geno) 
data_all$Cond <- factor(data_all$Cond,levels= Cond) 

Figure_S1H <- ggboxplot(data_all, x = "Cond", y = "NitrateContent.μmols.gFW.", color = "Cond",  palette =c("blue", "orange", "purple","red") ,add = "jitter") +  ggtitle("Nitrate Content ") + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed")   + ylab("NO3 (µmols/gFW)") + theme_bw() + geom_text(data=Hsd,aes(x=Cond,y= - 5,label=groups),vjust=0) + theme(axis.text.x = element_text(angle=90))   + xlab('')+ theme(legend.position="None")+ labs(colour='Condition')


Figure_S1H


 
```

### Figure_S1I: Total Carbon per plant

```{r Figure_S1I}

data_all <-  PhysiologicalParameter
 
Hsdcol <-   HSD.test(aov(TotalC ~  Cond, data = data_all[  data_all$Geno == "Col-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()  %>% mutate(Geno = "Col-0")
Hsdsvi <-   HSD.test(aov(TotalC ~  Cond, data = data_all[  data_all$Geno == "Cvi-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Cvi-0")
Hsdsha <-   HSD.test(aov(TotalC ~  Cond, data = data_all[ data_all$Geno == "Sha",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Sha")
Hsdbur <-   HSD.test(aov(TotalC ~  Cond, data = data_all[ data_all$Geno == "Bur-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno = "Bur-0")
Hsdtsu <-   HSD.test(aov(TotalC ~  Cond, data = data_all[ data_all$Geno == "Tsu-0",]), c("Cond" ), group=TRUE, alpha=0.05)$group %>% rownames_to_column("Cond")  %>% as.data.frame()   %>% mutate(Geno ="Tsu-0")

Hsd <- rbind(Hsdcol,Hsdsvi,Hsdsha,Hsdbur,Hsdtsu)
Hsd$Geno <- factor(Hsd$Geno,levels= Geno) 
sum <- ddply(data_all,.(Cond,Geno),summarise,n= length(d.13C.12C[!is.na(d.13C.12C)]))
sum$Geno <- factor(sum$Geno,levels= Geno) 
data_all$Geno <- factor(data_all$Geno,levels= Geno) 
data_all$Cond <- factor(data_all$Cond,levels= Cond) 

Figure_S1I <- ggboxplot(data_all, x = "Cond", y = "TotalC", color = "Cond",  palette =c("blue", "orange", "purple","red") ,add = "jitter") + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed")   + ylab("C (mg/rosette)")+ theme_bw()+ggtitle("Total Carbon per plant" )+ geom_text(data=Hsd,aes(x=Cond,y=0.5*min(TotalC),label=groups),vjust=0) +
theme(axis.text.x = element_text(angle=90)) + theme(legend.position="None") + xlab('') + labs(colour='Condition')

Figure_S1I
 
 
```



### Figure_S1: ALL except 
```{r Figure_S1}

FigureS1A_D <- ggarrange(Figure_S1A, Figure_S1B, Figure_S1C,Figure_S1D, 
          labels = c("A", "B", "C", "D"),  
          heights = c(1, 1,1,1),
          ncol = 1, nrow = 5)
ggsave("figures/Figure_S1A_D.svg", width = 16, height = 60, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S1A_D.jpeg", width = 16, height = 60, units = c("cm"), dpi = 600)


FigureS1E_I <- ggarrange(
          Figure_S1E, Figure_S1F,Figure_S1G,Figure_S1H,Figure_S1I,
          labels = c("E",'F','G','H','I'),  
          heights = c(.5,.5,.5,.5, .5),
          ncol = 1, nrow = 5)

ggsave("figures/Figure_S1E_I.svg", width = 16, height = 40, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S1E_I.jpeg", width = 16, height = 40, units = c("cm"), dpi = 600)

plot1 <- as.ggplot(FigureS1A_D)
plot2 <- as.ggplot(FigureS1E_I)

plot_grid(
  plot1, plot2, 
  align = "h", 
  nrow = 1, rel_widths = c(1, 1)
)

ggsave("figures/Figure_S1.jpg", width = 40, height = 80, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S1.svg", width = 40, height = 80, units = c("cm"), dpi = 600)
 
```

 

## Figure_S2

```{r Figure_S2}
PhenoData <- read.csv('data/Stressnet_PhenoData.csv') 

###PRA from D1-D23
PRAPval=NULL
example<- summary(aov(glm(PRA~ Geno*Experiment*Cond,data= PhenoData[PhenoData$Day==1,])))

for( i in unique(PhenoData$Day)){
  dataPRA <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(PRA~ Geno*Experiment*Cond,data= dataPRA)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("PRA"),i,Pvalj)
  PRAPval <- rbind(Pvalall, PRAPval)
}
colnames(PRAPval)<- c("Trait","Day",c(rownames(example[[1]][1:7,])))
####
RERiPval=NULL

for( i in unique(PhenoData$Day)){
  dataRERi <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(RERinstant~ Geno*Experiment*Cond, data= dataRERi)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("RERi3"),i,Pvalj)
  RERiPval <- rbind(Pvalall, RERiPval)
}
colnames(RERiPval)<- c("Trait","Day",c(rownames(example[[1]][1:7,])))

###Compactness D23
CompactnessiPval=NULL
for( i in unique(PhenoData$Day)){
  dataPvalCompactnessi <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(CompactnessPC~ Geno*Experiment*Cond,data= dataPvalCompactnessi)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("Compactness"),i,Pvalj)
  CompactnessiPval <- rbind(Pvalall, CompactnessiPval)
}

colnames(CompactnessiPval)<-c("Trait","Day",c(rownames(example[[1]][1:7,])))
###Hue D23
HueiPval=NULL
for( i in unique(PhenoData$Day)){
  dataPvalHuei <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(Hue_new~ Geno*Experiment*Cond,data= dataPvalHuei)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("Hue"),i,Pvalj)
  HueiPval <- rbind(Pvalall, HueiPval)
}

colnames(HueiPval)<-c("Trait","Day",c(rownames(example[[1]][1:7,])))

###Saturation D23
SaturationiPval=NULL
for( i in unique(PhenoData$Day)){
  dataPvalSaturationi <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(Saturation_new~ Geno*Experiment*Cond,data= dataPvalSaturationi)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("Saturation"),i,Pvalj)
  SaturationiPval <- rbind(Pvalall, SaturationiPval)
}

colnames(SaturationiPval)<-c("Trait","Day",c(rownames(example[[1]][1:7,])))

###Value D23
ValueiPval=NULL
for( i in unique(PhenoData$Day)){
  dataPvalValuei <- PhenoData[PhenoData$Day==i,]
  a<- summary(aov(glm(Value_new~ Geno*Experiment*Cond,data= dataPvalValuei)))
  Pvalj <- t(a[[1]][1:7,5])
  Pvalall <- data.frame(paste("Value"),i,Pvalj)
  ValueiPval <- rbind(Pvalall, ValueiPval)
}

colnames(ValueiPval)<-c("Trait","Day",c(rownames(example[[1]][1:7,])))

Pvalsummary<- rbind(PRAPval,RERiPval,CompactnessiPval,HueiPval,SaturationiPval,ValueiPval)
Pvaldata <- reshape::melt(Pvalsummary, id=c('Trait', 'Day'))

Pvaldata[,"DAS"] <- Pvaldata$Day +8
Pvaldata$variable <- gsub('Geno','Genotype',Pvaldata$variable)
Pvaldata$variable <- gsub('Cond','Condition',Pvaldata$variable)

Figure_S2 <- ggplot(Pvaldata, aes(DAS, -log(value), colour = variable) ) + geom_line() +ggtitle("Anova_Trait ~ Genotype * Experiment * Condition") + facet_wrap(Trait ~.,ncol = 3,scales = "fixed") + theme(axis.text.x = element_text(angle=90))  + ylab("-log(pvalue)") + geom_hline(yintercept = 3,color ="red") 
Figure_S2


ggsave("figures/Figure_S2.jpg", width = 20, height = 18, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S2.svg", width = 20, height = 18, units = c("cm"), dpi = 600)
 

 
```


## Figure_S3
```{r Figure_S3}

counts<- read.table("./data/RNA.counts.scaled.log2.tsv",check.names = F)
counts<- as.matrix(counts)
counts<- counts[,!(colnames(counts) %in% c("Bur.0_W1N1_P3ID34_b"))]
ymelt <- reshape2::melt(counts) %>% separate(Var2, c("Genotype","condition","phenoscope","rep"), sep = "_") 



ymelt <- ymelt %>%
  mutate(Genotype = case_when(
    Genotype == "Col.0" ~ "Col-0",
    Genotype == "Cvi.0" ~ "Cvi-0",
    Genotype == "Sha" ~ "Sha",
    Genotype == "Tsu.0" ~ "Tsu-0",
    Genotype == "Bur.0" ~ "Bur-0",
  )) %>% 
  mutate(condition = case_when(
    condition == "W0N0" ~ "W-N-",
    condition == "W0N1" ~ "W-N+",
    condition == "W1N0" ~ "W+N-",
    condition == "W1N1" ~ "W+N+",
  ))
ymelt$condition <- factor(ymelt$condition,levels= c("W+N+","W-N+","W+N-","W-N-")) 
ymelt$Genotype <- factor(ymelt$Genotype,levels= c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0"))  
ymelt$geneID <- factor(ymelt$Var1)  
ymelt$counts <-  ymelt$value  



bur0 <- readr::read_csv('data/Bur0_ALL.csv')%>% mutate(Geno = 'Bur-0')
col0 <- readr::read_csv('data/Col0_ALL.csv')%>% mutate(Geno = 'Col-0')
Cvi0 <- readr::read_csv('data/Cvi0_ALL.csv')%>% mutate(Geno = 'Cvi-0')
sha <- readr::read_csv('data/Sha_ALL.csv')%>% mutate(Geno = 'Sha')
Tsu0 <- readr::read_csv('data/Tsu0_ALL.csv')%>% mutate(Geno = 'Tsu-0')


dataframe <- rbind(bur0,Cvi0,col0,sha,Tsu0) %>% as.data.frame()
colnames(dataframe) <- gsub(' ','',colnames(dataframe) )

dataframe <- dataframe %>%
  mutate(geneID = case_when(
    Target == "act8" ~ "AT1G49240",
    Target == "gapd" ~ "AT1G16300",
    Target == "bor5" ~ "AT1G74810",
    Target == "cepd2" ~ "AT2G47880",
    Target == "cer1" ~ "AT1G02205",
    Target == "hho3" ~ "AT1G25550",
    Target == "nit4" ~ "AT5G22300",
    Target == "nrt2.5" ~ "AT1G12940",
    Target == "nudx" ~ "AT5G19470",
    Target == "pyl6" ~ "AT2G40330",
    Target == "roxy13" ~ "AT4G15680",
    Target == "roxy16" ~ "AT1G03020",
    Target == "roxy2" ~ "AT5G14070",
    Target == "roxy4" ~ "AT3G62950",
    Target == "roxy8" ~ "AT3G62960",
    Target == "tar" ~ "AT1G34060",
    TRUE ~ NA_character_   
  )) %>% 
  mutate(Cond = case_when(
    BiologicalGroup == "W0N0" ~ "W-N-",
    BiologicalGroup == "W0N1" ~ "W-N+",
    BiologicalGroup == "W1N0" ~ "W+N-",
    BiologicalGroup == "W1N1" ~ "W+N+",
  ))

# meanExpression 

# newqpcr <- dataframe[,c('Target','BiologicalGroup','MeanCq','CqSD','CqSEM','Geno','geneID','Cond')] %>%
#   group_by(Geno,BiologicalGroup) %>% do(mutate(., deltaCq= MeanCq- (MeanCq[Target == "act8"] + MeanCq[Target == "gapd"])/2  ))%>% ungroup()%>%
#   group_by(Geno,Target) %>% do(mutate(., deltaCon = deltaCq[BiologicalGroup=='W1N1']  ))  %>% ungroup()%>%
#   mutate(meanSEM = CqSEM  )%>%
#   mutate(reference = "ref")%>%
#   mutate(meanExpression = 2^-(deltaCq - deltaCon)  ) 
  
newqpcr <- dataframe[,c('Target','BiologicalGroup','Expression','ExpressionSEM','Geno','geneID','Cond')] %>%
           mutate(meanExpression = .$Expression  )%>%
           mutate(meanSEM = .$ExpressionSEM  )

  
newqpcr$Cond <- factor(newqpcr$Cond,levels= c("W+N+","W-N+","W+N-","W-N-")) 
newqpcr$Geno <- factor(newqpcr$Geno,levels= c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0"))  

gene_aliases <- read.table( "data/gene_aliases_20181231_summary.tsv",header = T)
newqpcr$geneSymbol <- gene_aliases$symbols[match(newqpcr$geneID ,gene_aliases$name)]


gene = "AT1G02205"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]
p1 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]) , x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p2 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw()  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT1G03020"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p3 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p4 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() +   ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT1G12940"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p5 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p6 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() +   ggtitle(title)+ theme(axis.text.x = element_text(angle=90))


gene = "AT1G25550"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p7 <-  ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() +  ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p8 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() +   ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT1G74810"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p9 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p10 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() +  ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT2G47880"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p11 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p12 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() +   ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT5G22300"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]


p13 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw()  + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p14 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw()   + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))


gene = "AT3G62960"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p15 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p16 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))


gene = "AT4G15680"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p17 <-  ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p18 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT2G40330"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p19 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p20 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT3G62950"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]


p21 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p22 <-   ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))
# 

gene = "AT5G19470"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]


p23 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p24 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT1G34060"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]


p25 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p26 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))

gene = "AT5G14070"
title = na.omit(newqpcr[newqpcr$geneID== gene ,])$geneSymbol[1]

p27 <- ggbarplot(  na.omit(newqpcr[newqpcr$geneID== gene ,]), x = "Cond", y = "meanExpression",  color = "Cond",palette = c("blue","orange","purple","red"),   lab.vjust = -1.6) +  facet_wrap(. ~ Geno  ,ncol = 5,scales = "fixed",drop = FALSE) +theme_bw()  +   geom_errorbar( aes(x=Cond, ymin=meanExpression-meanSEM, ymax=meanExpression + meanSEM), width=0.4, colour="black", alpha=0.9, linewidth=0.3) + theme_bw() + ylab("meanExpression ± meanSEM") + ggtitle(title) + theme(axis.text.x = element_text(angle=90))
p28 <-  ggboxplot(ymelt[ymelt$geneID==gene ,], x = "condition", y = "counts", color = "condition", add = "jitter",palette = c("blue","orange","purple","red"))  +  facet_wrap(. ~ Genotype,ncol = 5,scales = "fixed",drop = FALSE) +  rotate_x_text(90)  + ylab(" Normalized Counts") + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") + theme_bw() + ylab("counts ± meanSEM")  + ggtitle(title)+ theme(axis.text.x = element_text(angle=90))


(p1|p2)/(p3|p4)/(p5|p6)/(p7|p8)/(p9|p10) 
ggsave("figures/Figure_S3-1.jpg", width = 30, height = 60, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S3-1.svg", width = 30, height = 60, units = c("cm"), dpi = 600)

(p11|p12)/(p13|p14)/ (p15|p16)/(p17|p18)/ (p19|p20) 
ggsave("figures/Figure_S3-2.jpg", width = 30, height = 60, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S3-2.svg", width = 30, height = 60, units = c("cm"), dpi = 600)

(p21|p22)/(p23|p24)/(p25|p26)/(p27|p28) 
ggsave("figures/Figure_S3-3.jpg", width = 30, height = 60, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S3-3.svg", width = 30, height = 60, units = c("cm"), dpi = 600)


``` 


## Figure_S4

### Figure_S4B
```{r Figure_S4B1}

ecotype = "Cvi-0"
cond = 'W+N-'
resN <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N+'
resW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N-'
resNW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)
backgroundlist <- as.character(rownames(resW),rownames(resN),rownames(resNW))
```

```{r Figure_S4B2}
cond = 'W+N-'
dataset <- subset(resN,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S4Bup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S4Bdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S4B <-  Figure_S4Bup/Figure_S4Bdown
Figure_S4B
 
``` 

### Figure_S4C
```{r Figure_S4C}

cond = 'W-N-'

dataset <- subset(resNW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S4Cup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S4Cdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S4C <-  Figure_S4Cup/Figure_S4Cdown
Figure_S4C
 

 
``` 

### Figure_S4D
```{r Figure_S4D}

cond = 'W-N+'
dataset <- subset(resW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S4Dup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S4Ddown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S4D <-  Figure_S4Dup/Figure_S4Ddown
Figure_S4D
 
``` 

```{r Figure_S4D}
ggarrange(Figure_S4B, Figure_S4C, Figure_S4D, 
          labels = c("B",  "C",  'D'),  
          ncol = 3, nrow = 1)
ggsave("figures/Figure_S4B_D.jpg", width = 45, height = 35, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S4B_D.svg", width = 45, height = 35, units = c("cm"), dpi = 600)

``` 



## Figure_S5

### Figure_S5B
```{r Figure_S5B}

ecotype = "Sha"
cond = 'W+N-'
resN <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N+'
resW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N-'
resNW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)
backgroundlist <- as.character(rownames(resW),rownames(resN),rownames(resNW))

``` 

```{r Figure_S5B}
cond = 'W+N-'

dataset <- subset(resN,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S5Bup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S5Bdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S5B <-  Figure_S5Bup/Figure_S5Bdown
Figure_S5B
 
``` 

### Figure_S5C
```{r Figure_S5C}
cond = 'W-N+'

dataset <- subset(resNW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S5Cup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S5Cdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S5C <-  Figure_S5Cup/Figure_S5Cdown
Figure_S5C
 
``` 

### Figure_S5D
```{r Figure_S5D}

cond = 'W-N-'
dataset <- subset(resW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S5Dup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S5Ddown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S5D <-  Figure_S5Dup/Figure_S5Ddown
Figure_S5D
 
``` 

```{r Figure_S5D}
ggarrange(Figure_S5B, Figure_S5C, Figure_S5D, 
          labels = c("B",  "C",  'D'), align = "hv",
          ncol = 3, nrow = 1)
ggsave("figures/Figure_S5B_D.jpg", width = 45, height = 35, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S5B_D.svg", width = 45, height = 35, units = c("cm"), dpi = 600)

``` 


## Figure_S6

### Figure_S6B
```{r Figure_S6B1}

ecotype = "Tsu-0"
cond = 'W+N-'
resN <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N+'
resW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N-'
resNW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)
backgroundlist <- as.character(rownames(resW),rownames(resN),rownames(resNW))
``` 

```{r Figure_S6B2}
cond = 'W+N-'

dataset <- subset(resN,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S6Bup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S6Bdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S6B <-  Figure_S6Bup/Figure_S6Bdown
Figure_S6B
 
``` 

### Figure_S6C
```{r Figure_S6C}
cond = 'W-N-'

dataset <- subset(resNW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S6Cup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S6Cdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S6C <-  Figure_S6Cup/Figure_S6Cdown
Figure_S6C
 
``` 

### Figure_S6D
```{r Figure_S6D1}

cond = 'W-N+'
dataset <- subset(resW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S6Dup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S6Ddown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S6D <-  Figure_S6Dup/Figure_S6Ddown
Figure_S6D
 
``` 

```{r Figure_S6D2}
ggarrange(Figure_S6B, Figure_S6C, Figure_S6D, 
          labels = c("B",  "C",  'D'), align = "hv",
          ncol = 3, nrow = 1)
ggsave("figures/Figure_S6B_D.jpg",width = 45, height = 35, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S6B_D.svg", width = 45, height = 35, units = c("cm"), dpi = 600)

``` 


## Figure_S7

### Figure_S7B
```{r Figure_S7B1}

ecotype = "Bur-0"
cond = 'W+N-'
resN <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N+'
resW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)

cond = 'W-N-'
resNW <- read.table(paste0("data/Counts/",ecotype,'_',cond,"_refer_whole.tsv"),row.name="row",header=T)
backgroundlist <- as.character(rownames(resW),rownames(resN),rownames(resNW))
``` 

```{r Figure_S7B2}
cond = 'W+N-'

dataset <- subset(resN,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S7Bup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S7Bdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S7B <-  Figure_S7Bup/Figure_S7Bdown
Figure_S7B
 
``` 

### Figure_S7C
```{r Figure_S7C}
cond = 'W-N-'

dataset <- subset(resNW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S7Cup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S7Cdown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S7C <-  Figure_S7Cup/Figure_S7Cdown
Figure_S7C
 
``` 

### Figure_S7D
```{r Figure_S7D}

cond = 'W-N+'
dataset <- subset(resW,padj < 0.05 & abs(log2FoldChange) > 1)
geneList = dataset$log2FoldChange
names(geneList) = rownames(dataset)

geneList <- sort(geneList, decreasing = TRUE)

egoup <- enrichGO(names(geneList[geneList > 1]), keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                         ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egoup <- gofilter(egoup,level = 5)  %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio)))



egodown <- enrichGO(names(geneList[geneList < -1]),                 
                       keyType = "TAIR",
                        OrgDb = org.At.tair.db,
                          ont = "BP", 
                         universe = as.character(backgroundlist) , 
                         readable = TRUE,
                        pAdjustMethod = "BH",                 
                        pvalueCutoff = 0.05,                 
                        qvalueCutoff = 0.05)

egodown <- gofilter(egodown,level = 5) %>% mutate(., FoldEnrichment = parse_ratio(GeneRatio) / parse_ratio(BgRatio)) %>%  mutate(., richFactor = Count / as.numeric(sub("/\\d+", "", BgRatio))) 

 
Figure_S7Dup <- clusterProfiler::dotplot(egoup, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,'_Up')) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))   

Figure_S7Ddown <- clusterProfiler::dotplot(egodown, x="FoldEnrichment",showCategory=20) + ggplot2::ggtitle(paste0(ecotype,'_',cond,"_Down")) + scale_y_discrete(labels=function(y) str_wrap(y, width=35))

Figure_S7D <-  Figure_S7Dup/Figure_S7Ddown
Figure_S7D
 
``` 

```{r Figure_S7D}

ggarrange(Figure_S7B, Figure_S7C, Figure_S7D, 
          labels = c("B",  "C",  'D'), align = "hv",
          ncol = 3, nrow = 1)
ggsave("figures/Figure_S7B_D.jpg", width = 45, height = 35,units = c("cm"), dpi = 600)
ggsave("figures/Figure_S7B_D.svg", width = 45, height = 35,units = c("cm"), dpi = 600)

``` 

 
## Figure_S8
```{r Figure_S8}

ST002201_AN003604 <- read.delim("data/ST002201_AN003604.txt", header=FALSE)
metabolite_matrix<- ST002201_AN003604[2:137,] 
colnames(metabolite_matrix) <- ST002201_AN003604[1,]
metabolite_matrix[,-1] <- sapply(metabolite_matrix[,-1], FUN = as.numeric)

label <- ST002201_AN003604[139:nrow(ST002201_AN003604),]
colnames(label) <-  ST002201_AN003604[138,]
label$class <- gsub("^.*_","",label$metabolite_fullname)

 # symbol

data <- reshape::melt(metabolite_matrix,id = "metabolite_name")
data <- data %>% separate(variable,c("Geno","Cond","Exp","rep"),sep = "_")

Condition= c("W+N+","W-N+","W+N-","W-N-")

data$Geno <- factor( data$Geno,levels=c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) 
data$Cond <- factor(data$Cond,levels=Condition) 

p1 <-  ggboxplot(data[data$metabolite_name =="Citramalic acid",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Citramalic acid") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p2 <-  ggboxplot(data[data$metabolite =="Glutamic acid",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+ ggtitle("Glutamic acid") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20)) 

p3 <-  ggboxplot(data[data$metabolite =="Palmitoleic acid",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Palmitoleic acid") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p4 <-  ggboxplot(data[data$metabolite =="Phytol",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Phytol") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p5 <-  ggboxplot(data[data$metabolite =="Porphyrin-ring",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Porphyrin-ring") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p6 <-  ggboxplot(data[data$metabolite =="Rhamnose",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Rhamnose") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p7 <-  ggboxplot(data[data$metabolite =="Asparagine",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Asparagine") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p8 <-  ggboxplot(data[data$metabolite =="caffeic acid",],x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("caffeic acid") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p9 <-  ggboxplot(data[data$metabolite =="Ribulose-5-Phosphate",],x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Ribulose-5-Phosphate") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p10 <-  ggboxplot(data[data$metabolite =="Digalactosylglycerol",],x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Digalactosylglycerol") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p11 <-  ggboxplot(data[data$metabolite =="Sinigrin",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)") + xlab('')+   ggtitle("Sinigrin") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

p12 <-  ggboxplot(data[data$metabolite =="trans-Aconitic acid",], x = "Cond", y = "value", color = "Cond", palette =c("blue", "orange", "purple","red") ,add = "jitter") +      theme_bw()  + facet_wrap(. ~ Geno,ncol = 5,scales = "fixed") + theme(axis.text.x = element_text(angle=90)) + stat_compare_means(comparisons = list(c("W+N+","W-N+"),c("W+N+","W+N-"),c("W+N+","W-N-")), label = "p.signif",method = "t.test") +theme(legend.position = "none") +ylab("metabolite abundance/(ug/mgFW)")  + xlab('') +   ggtitle("trans-Aconitic acid") + theme(plot.title = element_text(size = 20),strip.text = element_text(size = 20))

Figure_S8A <- (p1|p2)/ (p3|p4)/ (p5|p6)
Figure_S8B <- (p7|p8)/ (p9|p10)/ (p11|p12)   

ggarrange(Figure_S8A, Figure_S8B,
          labels = c("A",  "B" ), align = "hv",
          ncol = 1, nrow = 2)

ggsave("figures/Figure_S8.jpg", width = 50, height = 70, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S8.svg", width = 50, height = 70, units = c("cm"), dpi = 600)

``` 

## Figure_S9
```{r Figure_S9}

ST002201_AN003604 <- read.delim("data/ST002201_AN003604.txt", header=FALSE)
metabolite_matrix<- ST002201_AN003604[2:137,]
label <- ST002201_AN003604[139:nrow(ST002201_AN003604),]
colnames(metabolite_matrix) <- ST002201_AN003604[1,]
colnames(label) <-  ST002201_AN003604[138,]
label$class <- gsub("^.*_","",label$metabolite_fullname)

data <- reshape::melt(metabolite_matrix,id = "metabolite_name")
data <- data %>% separate(variable,c("Geno","Cond","Exp","rep"),sep = "_")
Meta= levels(as.factor(data$metabolite_name))
Genotype=levels(as.factor(data$Geno))
data$value <- as.numeric(data$value)
wholemean <- plyr::ddply(data,.(metabolite_name,Geno,Cond) ,summarise, mean = median(value))
submean <- wholemean[wholemean$Cond =="W+N+",]

wholemean  <- merge(wholemean,submean[,!(colnames(submean) %in% c("Cond"))],by=c("metabolite_name","Geno"),all=T)
wholemean[,"foldchange"] <- round(foldchange(wholemean$mean.x, wholemean$mean.y),2)
wholemean[,"eco_cond"] <- paste0(wholemean$Geno,"_",wholemean$Cond)
wholemean <- wholemean %>% subset(., foldchange !=1) 
col_fun = colorRamp2(c(-10, 0, 10), c("blue", "white", "red"))

submat <- tidyr::spread(wholemean[,c("eco_cond","metabolite_name","foldchange")], eco_cond, foldchange,fill=0)
rownames(submat) <- submat$metabolite_name

# Phosphorylated intermediates. 
PCC <-  c("Sedoheptulose-7-Phosphate","Ribose-5-Phosphate","Glycerophosphoglycerol","Glucose-6-Phosphate","sn-Glycero-3-phosphate", "sn-Glycero-3-phosphoinositol", "D-Myo-Inositol-1-Phosphate","Phosphoric acid","Ribose-5-Phosphate","Ribulose-5-Phosphate")

smat <- submat[submat$metabolite_name %in% PCC,]

mat1 = as.matrix(smat[,c("Col-0_W+N-","Col-0_W-N-", "Col-0_W-N+", "Cvi-0_W+N-", "Cvi-0_W-N-", "Cvi-0_W-N+","Sha_W+N-","Sha_W-N-","Sha_W-N+", "Tsu-0_W+N-","Tsu-0_W-N-", "Tsu-0_W-N+","Bur-0_W+N-", "Bur-0_W-N-", "Bur-0_W-N+" )])

ht1 = Heatmap(mat1, name = "foldchange",cluster_columns = F, row_title = "A", 
              column_order =  colnames(mat1),col = col_fun)
 
# Tricarboxylic acid (TCA) cycle intermediates
OA <- c("Oxoglutaric acid","trans-Aconitic acid", "Citramalic acid","Citrate" ,"Fumaric acid","D-Malic acid", "Pyruvic acid","Ribonic acid","Salicylic acid","Shikimic acid","Succinic acid")
smat <- submat[submat$metabolite_name %in% OA,]

mat1 = as.matrix(smat[,c("Col-0_W+N-","Col-0_W-N-", "Col-0_W-N+", "Cvi-0_W+N-", "Cvi-0_W-N-", "Cvi-0_W-N+","Sha_W+N-","Sha_W-N-","Sha_W-N+", "Tsu-0_W+N-","Tsu-0_W-N-", "Tsu-0_W-N+","Bur-0_W+N-", "Bur-0_W-N-", "Bur-0_W-N+" )])
ht2 = Heatmap(mat1, name = "foldchange",cluster_columns = F, row_title = "B", 
              column_order =  colnames(mat1),col = col_fun)

 
#  Sugar metabolism intermediates
SUG <- label[label$class == "SUG",]$metabolite_name
smat <- submat[submat$metabolite_name %in% SUG,]

mat1 = as.matrix(smat[,c("Col-0_W+N-","Col-0_W-N-", "Col-0_W-N+", "Cvi-0_W+N-", "Cvi-0_W-N-", "Cvi-0_W-N+","Sha_W+N-","Sha_W-N-","Sha_W-N+", "Tsu-0_W+N-","Tsu-0_W-N-", "Tsu-0_W-N+","Bur-0_W+N-", "Bur-0_W-N-", "Bur-0_W-N+" )])

ht3 = Heatmap(mat1, name = "foldchange",cluster_columns = F, row_title = "C", 
              column_order =  colnames(mat1),col = col_fun)


# Amino acids
AA <- label[label$class == "AA",]$metabolite_name
smat <- submat[submat$metabolite_name %in% AA,]

mat1 = as.matrix(smat[,c("Col-0_W+N-","Col-0_W-N-", "Col-0_W-N+", "Cvi-0_W+N-", "Cvi-0_W-N-", "Cvi-0_W-N+","Sha_W+N-","Sha_W-N-","Sha_W-N+", "Tsu-0_W+N-","Tsu-0_W-N-", "Tsu-0_W-N+","Bur-0_W+N-", "Bur-0_W-N-", "Bur-0_W-N+" )])

ht4 = Heatmap(mat1, name = "foldchange",cluster_columns = F, row_title = "D", 
              column_order =  colnames(mat1),col = col_fun)

jpeg('figures/Figure_S9.jpg', units="cm", width=25, height=30, res=600)

Figure_S9 <- ht1 %v% ht2 %v% ht3 %v% ht4  
Figure_S9 

dev.off()
 
 
``` 

## Figure_S10
```{r Figure_S9}



``` 

## Figure_Sup
```{r Figure_S9}

ST002201_AN003604 <- read.delim("data/ST002201_AN003604.txt", header=FALSE)
metabolite_matrix<- ST002201_AN003604[2:137,]
label <- ST002201_AN003604[139:nrow(ST002201_AN003604),]
colnames(metabolite_matrix) <- ST002201_AN003604[1,]
colnames(label) <-  ST002201_AN003604[138,]
label$class <- gsub("^.*_","",label$metabolite_fullname)
metabolite_matrix[,-1] <- sapply(metabolite_matrix[,-1], FUN = as.numeric)

data <- reshape::melt(metabolite_matrix,id = "metabolite_name")
data <- data %>% separate(variable,c("Geno","Cond","Exp","rep"),sep = "_")
data$value <- as.numeric(data$value)
Condition= c("W+N+","W-N+","W+N-","W-N-")

data$Geno <- factor( data$Geno,levels=c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) 
data$Cond <- factor(data$Cond,levels=Condition) 
ggplot(data, aes(x = value,fill = Cond)) + geom_histogram(binwidth = 0.05) + ggtitle('ST002201_AN003604_data_distribution') +  facet_grid( ~ Geno)  
ggsave("figures/Figure_S10.jpg", width = 50, height = 35, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S10.svg", width = 50, height = 35, units = c("cm"), dpi = 600)
 
``` 


## Figure_sup
```{r Figure_S9}

counts<- read.table(paste0("data/Counts/merge_counts_scaled_log2.tsv"),check.names = F)
counts<- counts[,!(colnames(counts) %in% c("Bur.0_W1N1_P3ID34_b"))] 

data <-   reshape::melt(counts,id ='GeneID')
data <- data %>% separate(variable,c("Geno","Cond","Exp","rep"),sep = "_")
Condition= c("W+N+","W-N+","W+N-","W-N-")

data$Geno <- factor( data$Geno,levels=c("Col-0","Cvi-0","Sha","Tsu-0","Bur-0")) 
data$Cond <- factor(data$Cond,levels=Condition) 
ggplot(data, aes(x = value,fill = Cond)) + geom_histogram(binwidth = 0.05) + ggtitle('Gene_expression_data_distribution') +  facet_grid( ~ Geno)  
ggsave("figures/Figure_S10.jpg", width = 50, height = 35, units = c("cm"), dpi = 600)
ggsave("figures/Figure_S10.svg", width = 50, height = 35, units = c("cm"), dpi = 600)

 
``` 
